src/foam/u2/FloatView.js

```diff
@@ -112,10 +112,30 @@ foam.CLASS({
     function viewListenerFn() {
       var data = this.data$;
       var view = this.attrSlot(null, this.onKey ? 'input' : null);
+
+      this.protectFromFeedback(() => {
+        const el = this.el_();
+
+        let pos = el.selectionStart;
+        // new text will be selected if it immediately follows the caret
+        let selectNewText = pos == el.value.length;
+
+        // check bounds on data update and set to boundary values if out of bounds
+        data.set(this.textToData(
+          foam.Number.clamp(this.min, view.get(), this.max)));
+
+        // preserve caret location, maybe selecting new text
+        el.selectionStart = Math.min(pos, el.value.length);
+        if ( ! selectNewText ) el.selectionEnd = el.selectionStart;
+      });
+    },
+
+    function protectFromFeedback (fn) {
```

Comment:
Is this useful?
https://github.com/kgrgreer/foam3/blob/3eafc7f2694ace889d94ee2e785a4fc86e45ae9d/src/foam/util/DeFeedback.js